packages/did/src/DidDetails/DidDetails.ts

```diff
@@ -7,248 +7,100 @@
+}
 
-  /**
-   * Compute the full URI (did:kilt:<identifier>#<key_id> for a given DID key <key_id>.
-   *
-   * @param keyId The key ID, without the leading subject's DID prefix.
-   *
-   * @returns The full public key URI, which includes the subject's DID and the provided key ID.
-   */
-  public assembleKeyUri(keyId: DidKey['id']): DidResourceUri {
-    return assembleKeyUri(this.uri, keyId)
-  }
+/**
+ * Returns a service endpoint with a given id, if associated with this Did.
+ *
+ * @param did The DID data.
+ * @param id Endpoint id (not the full endpoint uri).
+ * @returns The respective endpoint data or undefined.
+ */
+export function getEndpoint(
+  did: Pick<DidDetails, 'service'>,
+  id: DidServiceEndpoint['id']
+): DidServiceEndpoint | undefined {
+  return did.service?.find((endpoint) => endpoint.id === id)
+}
 
-  /**
-   * Generate a signature over the provided input payload, either as a byte array or as a HEX-encoded string.
-   *
-   * @param payload The byte array or HEX-encoded payload to sign.
-   * @param sign The sign callback to use for the signing operation.
-   * @param keyId The key ID to use to generate the signature.
-   *
-   * @returns The resulting [[DidSignature]].
-   */
-  public async signPayload(
-    payload: Uint8Array | string,
-    sign: SignCallback,
-    keyId: DidVerificationKey['id']
-  ): Promise<DidSignature> {
-    const key = this.getKey(keyId)
-    if (!key || !isVerificationKey(key)) {
-      throw new SDKErrors.DidError(
-        `Failed to find verification key with ID "${keyId}" on DID "${this.uri}"`
-      )
-    }
-    const alg = getSigningAlgorithmForVerificationKeyType(
-      key.type as VerificationKeyType
+/**
+ * Generate a signature over the provided input payload, either as a byte array or as a HEX-encoded string.
+ *
+ * @param did The DID data.
+ * @param payload The byte array or HEX-encoded payload to sign.
+ * @param sign The sign callback to use for the signing operation.
+ * @param keyId The key ID to use to generate the signature.
+ *
+ * @returns The resulting [[DidSignature]].
+ */
+export async function signPayload(
+  did: Partial<DidDetails> & Pick<DidDetails, 'authentication' | 'uri'>,
+  payload: Uint8Array | string,
+  sign: SignCallback,
+  keyId: DidVerificationKey['id']
+): Promise<DidSignature> {
+  const key = getKey(did, keyId)
+  if (!key || !isVerificationKey(key)) {
+    throw new SDKErrors.DidError(
+      `Failed to find verification key with ID "${keyId}" on DID "${did.uri}"`
     )
-    if (!alg) {
-      throw new SDKErrors.DidError(
-        `No algorithm found for key type "${key.type}"`
-      )
-    }
-    const { data: signature } = await sign({
-      publicKey: key.publicKey,
-      alg,
-      data: Crypto.coToUInt8(payload),
-    })
-    return {
-      keyUri: this.assembleKeyUri(key.id),
-      signature: u8aToHex(signature),
-    }
+  }
+  const alg = getSigningAlgorithmForVerificationKeyType(
```

Comment:
I am uncertain how much we should doubt TS in the code. Here it tells us that the algorithm will always be defined. Why double-check here but not in all other places?

Reply:
TypeScript can maybe be trusted, but not necessarily JavaScript. Since the key is gotten via an internal function and checked for existence, it is fine in this case, though.

Reply:
> Since the key is gotten via an internal function and checked for existence, it is fine in this case, though.

This is why the question surprised me.